argus/shell.py

```diff
@@ -0,0 +1,261 @@
+                        help="Make sepatate log files for each scenario.")
+    parser.add_argument("-g", "--git_command",
+                        help="The git command that will test the "
+                             "Cloudbase-Init patch.")
+    parser.add_argument("-m", "--msi", help="The msi to use.")
+    return parser
+
+
+def _prepare_enviroment(local, directory):
+    """Prepare the temp enviroment."""
+    os.mkdir(os.path.join(directory, "ci"))
+
+    if local:
+        local = os.path.abspath(local)
+        os.link(os.path.join(local, ".testr.conf"),
+                os.path.join(directory, ".testr.conf"))
+
+        os.link(os.path.join(local, "ci", "tests.py"),
+                os.path.join(directory, "ci", "tests.py"))
+    else:
+        # Download the necessary items
+        download_argus_resource(".testr.conf", os.path.join(
+            directory, ".testr.conf"))
+
+        download_argus_resource("ci/tests.py", os.path.join(
+            directory, "ci", "tests.py"))
+
+    # Create a new repository
+    proc = subprocess.Popen(["testr", "init"], cwd=directory, close_fds=True)
+    proc.wait()
+
+
+def _prepare_config(separate, resources, flavor,
+                    git_command, msi,
+                    directory, image_id):
+    """Prepare the argus config file."""
+
+    conf = six.moves.configparser.SafeConfigParser()
+    conf.add_section("argus")
+    conf.add_section("openstack")
+
+    conf.set("argus", "output_directory", os.path.join(directory, "output"))
+    conf.set("argus", "argus_log_file", os.path.join(directory, "argus.log"))
+
+    if resources:
+        conf.set("argus", "resources", str(resources))
+
+    if separate:
+        conf.set("argus", "individual_logs", str(separate))
+
+    if flavor:
+        conf.set("openstack", "flavor_id", str(flavor))
+
+    conf.set("openstack", "image_ref", str(image_id))
+    conf.set("argus", "git_command", str(git_command))
+    conf.set("argus", "patch_install", str(msi))
+
+    config_path = os.path.join(directory, "argus.conf")
+    with open(config_path, 'w') as file_handle:
+        conf.write(file_handle)
+
+    return config_path
+
+
+def _start_testr(parallel, tests, directory):
+    """Start running the tests."""
+
+    cmd = ["testr", "run", str(tests)]
+    if parallel:
+        cmd.append("--parallel")
+        cmd.append("--concurrency={}".format(parallel))
+
+    process = subprocess.Popen(cmd, cwd=directory, close_fds=True)
+
+    return process
+
+
+def main():
+    """The main entry point."""
+    parser = _prepare_argument_parser()
+    args = parser.parse_args(sys.argv[1:])
+    images_id = _get_images(args.image_id, args.image_name)
+
+    base_directory = _get_base_directory(args.directory)
+
+    print("[INFO] Starting at {}".format(base_directory))
+    sys.stdout.flush()
+
+    processes = []
+
+    for image_id, image_name in images_id:
+        directory = os.path.join(base_directory, str(image_name))
+        os.mkdir(directory)
```

Comment:
Please check beforehand, `if not os.path.exists(directory): os.mkdir(directory)`.